cpp/src/join/mixed_join_semi.cu

```diff
@@ -14,9 +14,9 @@
  * limitations under the License.
  */
 
-#include "join_common_utils.cuh"
-#include "join_common_utils.hpp"
-#include "mixed_join_kernels_semi.cuh"
+#include "join/join_common_utils.cuh"
+#include "join/join_common_utils.hpp"
+#include "join/mixed_join_kernels_semi.cuh"
```

Comment:
changes not needed
```suggestion
#include "join_common_utils.cuh"
#include "join_common_utils.hpp"
#include "mixed_join_kernels_semi.cuh"
```

Reply:
I had added this because of a previous comment from @bdice. https://github.com/rapidsai/cudf/pull/16321#discussion_r1688315786

Reply:
I see that the developer guide only mentions about avoiding relative paths not in the same directory -- `Avoid relative paths with .. when possible. Paths with .. are necessary when including (internal) headers from source paths not in the same directory as the including file, because source paths are not passed with -I`. I can revert this to not include the `join` directory.

Reply:
Apologies for the inconsistent guidelines provided by the libcudf team. We will work to improve clarity on inclusion guidelines moving forward. For this PR, to avoid unnecessary back-and-forth, we can leave it as it is.